addons/l10n_sa_edi/wizard/account_move_reversal_views.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="view_account_move_reversal_inherit_l10n_sa_edi" model="ir.ui.view">
+            <field name="name">account.move.reversal.form.inherit.l10n_sa_edi</field>
+            <field name="inherit_id" ref="account.view_account_move_reversal"/>
+            <field name="model">account.move.reversal</field>
+            <field name="arch" type="xml">
+                <field name="reason" position="replace">
+                    <field name="country_code" invisible="1"/>
+                    <field name="reason" string="Reason" invisible="move_type == 'entry' and country_code != 'SA'" required="move_type != 'entry' or country_code == 'SA'"/>
```

Comment:
```suggestion
                    <field name="reason" string="Reason" invisible="move_type == 'entry' and country_code != 'SA'" required="country_code == 'SA'"/>
```
That makes it required for other localization, which is not intended I imagine ? :thinking: 
For non-SA country it should be visible if it's not an entry but not required. For SA country it should be visible and required.